sampleapp/src/main/scala/de/up/hpi/informationsystems/sampleapp/dactors/Cart.scala

```diff
@@ -54,8 +65,123 @@ class Cart(id: Int) extends Dactor(id) {
     Map("cart_info" -> CartInfo) ++ Map("cart_purchases" -> CartPurchases)
 
   override def receive: Receive = {
-    case AddItems.Request(_, _) => sender() ! AddItems.Failure(new NotImplementedError)
+    case AddItems.Request(orders, customerId) =>
+      addItems(orders, customerId).map(res => res match {
+        case Success(newRecords) => sender() ! AddItems.Success(newRecords.head.get(CartPurchases.sessionId).get)
+        case Failure (e) => sender() ! AddItems.Failure(e)
+      })
+
     case Checkout.Request(_) => sender() ! Checkout.Failure(new NotImplementedError)
   }
 
+  def addItems(orders: Seq[AddItems.Order], customerId: Int): Future[Try[Seq[Record]]] = {
+    // ask for prices of orders
+    // i_id, i_price, i_min_price
+    val prices = Future.sequence(orders
+      .groupBy(order => order.sectionId)
+      .map(askStoreSectionForPrices))
+
+    // c_g_id
+    val customer_group_id = askCustomerForGroupId(customerId)
+
+    // collect i_ids, group_id to query for fixed_disc
+    val i_ids_and_group_id = for {
+      prices <- prices
+      c_g_id <- customer_group_id
+    } yield (prices.flatten.map(ppr => ppr.i_id), c_g_id)
+
+    // i_id, fixed_disc
+    val fixed_disc = i_ids_and_group_id.flatMap(tuple => {
+      val (i_ids, group_id) = tuple
+      askGroupManagerForDiscount(i_ids.toSeq, group_id)
+    })
```

Comment:
**Style - this is subjective**

There is an alternative syntax for this, you may consider:

```scala
val fixed_disc = i_ids_and_group_id.flatMap{
  case (i_ids, group_id) => askGroupManagerForDiscount(i_ids.toSeq, group_id)
}
```